### PR TITLE
Update rke types for cloud providers

### PR DESCRIPTION
**Pull request by galal-hussein**

### CODE REVIEW

File: apis/management.cattle.io/v3/rke_types.go [hunk 1]
```diff
@@ -337,14 +337,146 @@ type PortCheck struct {
...
+	// AWSCloudProvider
+	AWSCloudProvider *AWSCloudProvider `yaml:"awsCloudProvider,omitempty" json:"awsCloudProvider,omitempty"`
+	// AzureCloudProvider
+	AzureCloudProvider *AzureCloudProvider `yaml:"azureCloudProvider,omitempty" json:"azureCloudProvider,omitempty"`
+	// OpenstackCloudProvider
+	OpenstackCloudProvider *OpenstackCloudProvider `yaml:"openstackCloudProvider,omitempty" json:"openstackCloudProvider,omitempty"`
+	// VsphereCloudProvider
+	VsphereCloudProvider *VsphereCloudProvider `yaml:"vsphereCloudProvider,omitempty" json:"vsphereCloudProvider,omitempty"`
+	// CustomCloudProvider
+	CustomCloudProvider string `yaml:"customCloudProvider,omitempty" json:"customCloudProvider,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/rke_types.go]: what is custom cloud provider? does it come with 0 options?

Comment (galal-hussein) [reply apis/management.cattle.io/v3/rke_types.go]: Custom cloud provider is just multi line string, where the user can put a custom config file

Comment (alena1108) [reply apis/management.cattle.io/v3/rke_types.go]: Can we add it to the comments?

Comment (galal-hussein) [reply apis/management.cattle.io/v3/rke_types.go]: sure

**Pull request merged**